src/use-previous/index.ts

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react'
+
+export const usePrevious = <T>(value: T): T => {
+  const ref = React.useRef(value)
```

Comment:
Different from original implementation. Adding `value` as initial value for ref to remove nullable type. Something we had discussed as an improvement here + typescript actually complains in line 7 that `type T is not assignable to type undefined`, flow somehow missed this.

Reply:
Actually decided to leave this change of implementation for later to reduce the changed variables and was able to type it so that TS was happy.